cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -813,6 +864,60 @@ func (m *McmManager) getMachineDeploymentUntilDeadline(mdName string, retryInter
 	}
 }
 
+// getMachineUntilDeadline returns error only when fetching the machine has been failing consequently and deadline is crossed
+func (m *McmManager) getMachineUntilDeadline(mcName string, retryInterval time.Duration, deadline time.Time) (*v1alpha1.Machine, error) {
+	for {
+		mc, err := m.machineLister.Machines(m.namespace).Get(mcName)
+		if err != nil && time.Now().Before(deadline) {
+			klog.Warningf("Unable to fetch MachineDeployment object %s, Error: %s, will retry in %s", mcName, err, retryInterval)
+			time.Sleep(retryInterval)
```

Comment:
please do not use `time.Sleep` as it prevents the go runtime to reschedule this go-routine. A thread is blocked and cannot be used for another go-routine. 
We have already coded a retry utility in DWD please use that instead or you can also use the one that comes natively with client-go (https://pkg.go.dev/k8s.io/client-go/util/retry). Same comment for all functions where you are doing the retry

Reply:
https://pkg.go.dev/k8s.io/client-go/util/retry also uses `time.Sleep` in its implementation. In DWD, for `retryOnError` we use `time.Sleep`, we use channels and `time.After` as well in other methods, but I think it will be the same thing.

Reply:
Ok k8s utility is using time.Sleep then it is a bad coding choice. At least in DWD we use time.After and that is better as go scheduler is now able to move out this routine from scheduling onto to a thread till it gets an event on this channel at which time the go-routine is scheduled back. However we should also improve in DWD and not use time.After but create a ticker and stop the ticker in the defer call.

Reply:
addressed in [818628e](https://github.com/gardener/autoscaler/pull/215/commits/818628ebcc0a9dd831c382efafaf39fad1a86753)